src/extensions/askUserForParam.ts

```diff
@@ -150,16 +157,42 @@ export async function askForParam(args: {
           },
           message:
             "Which options do you want enabled for this parameter? " +
-            "Press Space to select, then Enter to confirm your choices. " +
-            "You may select multiple options.",
+            "Press Space to select, then Enter to confirm your choices. ",
           choices: convertExtensionOptionToLabeledList(paramSpec.options as ParamOption[]),
         });
         valid = checkResponse(response, paramSpec);
         break;
       case ParamType.SECRET:
-        response = args.reconfiguring
-          ? await promptReconfigureSecret(args.projectId, args.instanceId, paramSpec)
-          : await promptCreateSecret(args.projectId, args.instanceId, paramSpec);
+        while (!secretLocations.length) {
+          secretLocations = await promptOnce({
+            name: "input",
+            type: "checkbox",
+            message:
+              "Where would you like to store your secrets? You must select at least one value",
+            choices: [
+              {
+                checked: true,
+                name: "Google Cloud Secret Manager",
+                // return type of string is not actually enforced, need to manually convert.
+                value: SecretCreationAction.CLOUD.toString(),
+              },
+              {
+                checked: false,
+                name: "Local file (Only used by Firebase Emulator)",
+                value: SecretCreationAction.LOCAL.toString(),
+              },
+            ],
+          });
+        }
+        if (secretLocations.includes(SecretCreationAction.CLOUD.toString())) {
```

Comment:
Optional: This chunk from l187 to l193 could be good to separate into its own function, for readability.